src/apps/frontend/components/header/user-profile-snippet.component.tsx

```diff
@@ -22,7 +22,10 @@ const UserProfileSnippet: React.FC<DropdownUserProps> = ({
   const trigger = useRef<HTMLAnchorElement>(null);
   const dropdown = useRef<HTMLDivElement>(null);
 
-  const username = account?.name || account?.username || 'Guest';
+  let username = account?.username || 'Guest';
+  if (account?.firstName) {
+    username = `${account?.firstName} ${account?.lastName}`;
+  }
```

Comment:
I don't understand why account would be nil here. I would expect the user profile snippet to show only if there is an account.

Moving forward, with this change, I expect first and last name to be available so there is not a whole lot of need to do if else here.

I would also add `displayName` in `Account` object on client side which does this concatenation to better encapsulate the logic